clickhouse_sqlalchemy/ext/clauses.py

```diff
@@ -29,6 +31,19 @@ def sample_clause(element):
         return SampleParam(None, element, unique=True)
 
 
+class LimitByClause:
+
+    def __init__(self, by_clauses, offset, limit):
```

Comment:
According to docs `LIMIT BY` has following syntax:

```sql
    LIMIT [offset_value, ]n BY expressions
    LIMIT n OFFSET offset_value BY expressions
```

It's better to keep parameters order the same: `limit, offset, *by_clauses`.

Reply:
I can't recall any case where I needed `offset` parameter. It's used exceptionally rarely, so making it mandatory would be a strange decision.

Reply:
Or do you mean the order in `LimitByClause.__init__` only, but not `{Query,Select}.limit_by` methods? This way it looks ok to me.

Reply:
Yes, we can use `.limit_by(limit, *by_expressions, offset=None)` in `Query`/`Select`. Currently `limit` after `offset=None` produces syntax error on old python versions.

Reply:
They're keyword-only parameters, so "limit after offset" sounds a bit odd. Yes, I forgot you're tied to 2.7. Is it still reasonable to support the version half a year after end of support? If so, I suggest the following alternatives:
```python
def limit_by(by_clauses, limit, offset=None):
    ...
# .limit_by([col1, col2], limit=1)
# .limit_by([col1, col2], 10, 1)
```
or
```
def limit_by(limit, offset=None, by=None):
   ...
# .limit_by(1, by=[col1, col2])
# .limit_by(1, offset=10, by=[col1, col2])
```

Breaking close related parameters `limit` and `offset` doesn't look good.

Reply:
> Yes, I forgot you're tied to 2.7. Is it still reasonable to support the version half a year after end of support?

It looks like there are still reasonable amount of downloads for 2.x version: https://pypistats.org/packages/clickhouse-sqlalchemy.

I think one of next SA releases will drop python 2 support. It's not so hard to keep backwards compatibility right now.

OK, let's pick `def limit_by(by_clauses, limit, offset=None):`.

Reply:
I've changed the interface according to your decision. But tests fail anyway, since `mock` lib (used by `responses`, which is used in `tests/drivers/http/test_transport.py`) is not compatible to Python 2.7